docs/SDK_SPEC.md

```diff
@@ -898,9 +900,31 @@ The execution of a Function, the returned data, and the tooling available to a D
 
 ## 4.5. Introspection Requests
 
-An SDK MAY offer the ability to respond to an Introspection Request, a `GET` request intended to be used as a health check, ensuring that an HTTP request can be correctly handled by an SDK.
+An SDK MUST respond to an Introspection Request, a `GET` request intended to be used as a health check, ensuring that an HTTP request can be correctly handled by an SDK.
 
-If an SDK chooses to respond to this method, it MUST NOT return any sensitive data.
+If the request signature validation fails, then the SDK MUST NOT return any sensitive data. The response schema MUST be:
+
+```ts
+{
+	function_count: number
+	has_event_key: boolean
+	has_signing_key: boolean
+  mode: "cloud" | "dev"
+}
+```
```

Comment:
Can we rephrase this to say that it must contain at least these fields? For example, as you know in TS it's useful to also see if the mode is explicit or was inferred.

Actually, perhaps we namespace any custom keys to make sure an SDK doesn't take a name we want to later use generically?
```suggestion
If the request signature validation fails, then the SDK MUST NOT return any sensitive data. The response schema MUST be:

```ts
{
  function_count: number;
  has_event_key: boolean;
  has_signing_key: boolean;
  mode: "cloud" | "dev";
  metadata?: Record<string, any>;
}
```
```